packages/grafana-ui/src/components/Segment/SegmentAsync.story.tsx

```diff
@@ -202,4 +204,59 @@ export const HtmlAttributes = () => {
   );
 };
 
+export const Basic: ComponentStory<React.ComponentType<SegmentAsyncProps<string>>> = (
+  args: SegmentAsyncProps<string>
+) => {
+  const [value, setValue] = useState(args.value);
+
+  const props: SegmentAsyncProps<string> = {
+    ...args,
+    value,
+    loadOptions: async (query = '') => {
+      action('loadOptions fired')({ query });
+      const result = await loadOptions(options);
+      if (query) {
+        return result.filter((data) => data.label?.includes(query));
+      }
+      return loadOptions(options);
+    },
+    onChange: ({ value }) => {
+      setValue(value);
+      action('onChange fired')(value);
+    },
+    onExpandedChange: (expanded) => action('onExpandedChange fired')({ expanded }),
```

Comment:
Pass `noOptionMessageHandler` handler to props to make it clear in Actions tab when this handler is called.

Reply:
@JoaoSilvaGrafana, I have solved this within [bce0928](https://github.com/grafana/grafana/pull/55264/commits/bce0928655f9b326bde5e66f4bb0c24d8e43795b), feel free to review this PR when you have a chance, thank you!